encoding/thrift/outbound.go

```diff
@@ -193,19 +193,24 @@ func (c thriftClient) CallOneway(ctx context.Context, reqBody envelope.Enveloper
 }
 
 func (c thriftClient) buildTransportRequest(reqBody envelope.Enveloper) (*transport.Request, protocol.Protocol, error) {
+	thriftEnvelope := "true"
 	proto := c.p
 	if !c.Enveloping {
 		proto = disableEnvelopingProtocol{
 			Protocol: proto,
 			Type:     wire.Reply, // we only decode replies with this instance
 		}
+		thriftEnvelope = "false"
 	}
 
 	treq := transport.Request{
 		Caller:    c.cc.Caller(),
 		Service:   c.cc.Service(),
 		Encoding:  Encoding,
 		Procedure: procedure.ToName(c.thriftService, reqBody.MethodName()),
+		TransportHeaders: map[string]string{
+			"Thrift-Envelope": thriftEnvelope,
```

Comment:
Regardless of if this is in `Headers` or `TransportHeaders`, this should be a constant.